### PR TITLE
Refactor 2016 04 12

### PR DESCRIPTION
**Pull request by phadej**

### CODE REVIEW

File: src/Main.hs [hunk 1]
```diff
@@ -190,15 +197,15 @@ findUserNick users uid =
 
 lookupFlows :: FlowMap -> ParamName Organisation -> [ParamName Flow]
 lookupFlows flowsMap org =
-  maybe [] id $ HM.lookup org flowsMap
+  maybe [] (L.sort . NE.toList) $ HM.lookup org flowsMap
```
Comment (daniellandau) [inline src/Main.hs]: If you don't sort, the order they come from the API is what the user has dragged them to. I think that could be a sensible order to keep, for example if passing the output to `head` or further filtering/processing with `grep`/`awk`/etc

File: src/Main.hs [hunk 2]
```diff
@@ -111,23 +113,28 @@ paramArgument m = mkParamName <$> strArgument m
...
 optsParser :: Maybe AuthToken -> Parser Opts
 optsParser maybeToken =
   Opts <$> authTokenParser maybeToken
-       <*> switch (long "offline" <> help "Consider only already downloaded logs")
+       <*> switch (short 'o' <> long "offline"     <> help "Consider only already downloaded logs")
        <*> switch (short 'i' <> long "ignore-case" <> help "Perform case insensitive matching")
-       <*> switch (long "all" <> help "Search over all flows")
-       <*> switch (long "show-date" <> help "Show date in search results")
-       <*> (optional $ strOption (long "by" <> metavar "user" <> help "Only posts by user"))
+       <*> switch (short 'd' <> long "show-date  " <> help "Show date in search results")
```
Comment (daniellandau) [inline src/Main.hs]: `"show-date  "` shouldn't have spaces, i.e.:

```
$ stack exec flowdock-grep -- --show-date   --all futurice whatever
   Invalid option `--show-date`
$ stack exec flowdock-grep -- "--show-date  " --all futurice whatever
   normal output
```

**Comment by daniellandau** [comment]
Other than the two things :+1:. Option parsing might warrant a test or two at this point.

**Comment by phadej** [comment]
Let's write tests when we on the mood. hmm...

**Pull request merged**